### ISSUE TITLE
Add/Remove Page Breaks (to match Add/Remove System Breaks)

### ISSUE THREAD

**Issue by wizofaus** [musescore/MuseScore#23792]
### Your idea

Add a command like the existing "Add/Remove System Breaks" but for Page Breaks.

### Problem to be solved

Once you have your layout the way you want it in terms of page breaks etc., having the ability to fix them in place so that further edits don't cause systems to move around between pages would be super helpful.  Also in some cases you really do want a fixed number of systems per page. 

### Additional context

Could potentially expand existing command, but I think it's sufficiently independent to justify a separate one.  Obviously when page breaks are added, any existing system breaks *could* be removed, but possibly shouldn't be just in case you later decide you want more systems per page but to keep the exist system breaks.

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by DaddyLudwig** [comment]
I would appreciate this too, cause I often want all my pages to have the same number of systems, like 6 systems a page in a Beethoven piano sonata for example. And while I often find that it settles around the system count per page that I want once I set the staff space property how I want it in Format -> Page Settings(I generally start with 1.55 mm and then if there are some pages with fewer systems than I want, I bring it down a little), sometimes I find that even after inputting all the notes of the movement for instance, there's a couple pages that have more systems than I want. So I end up having to select the last measure of the system I want to be the last and do Ctrl + Enter.

**Comment by rpatters1** [comment]
I totally agree with the need for the Remove part of this feature request. The same thing can be accomplished with Select similar, but it is not as easy nor as straightforward as a Remove command.

The challenge I see for Add, which for some reason seems a worse challenge to me than for system breaks, is that there is no way to guarantee that each page has exactly the requested number of systems. All you can guarantee is that they have *at most* that many systems. If the `x` systems on page `y` require too much vertical space, MuseScore will split them into two pages and result in a terrible layout for those two pages.

I suppose adding hard page breaks at least guarantees that changes will only affect the layout of the single page where the change is. But I think this might be a much more useful feature if the number of systems could be enforced to take priority over the collision avoidance logic.

I have opened #25107 to request features to permit more control over the final effort of polishing the page layout. It includes a request for forcing measures-per-system and systems-per-page. But perhaps that runs very counter to the design philosophy of MuseScore, so it could be a forlorn hope. (And maybe that's a case of be careful what you wish for: idk.)

**Comment by wizofaus** [comment]
TBH I hadn't even really considered the "Remove" part, though admittedly just recently after having added system/page breaks to a score to ensure the layout remained stable, I decided the work needed a quite a few new measures, and I will say once you do that existing break objects that occur *after* the inserted measures almost certainly don't make sense any more.

"Forcing systems-per-page" would be part of the functionality added here (just like you can already force measures-per-system using "Add/Remove System Breaks").  But I guess you mean you want it to be a score setting "always have n systems per page, no matter what measures I add or remove".   Maybe there's a workflow for which that would make sense, though I personally avoid fixing the layout until I'm pretty much ready to export the PDFs.

**Comment by DaddyLudwig** [comment]
I often scatter the formatting adjustments through the process of making the score. Like, I'll do an initial notch down of staff space from 1.75 mm to 1.55 mm, and then if I see a page with fewer systems than the rest, I'll do little increments down, like not even a full down arrow button press, smaller than that, to try to even things out (more likely in orchestral works (specifically those for a smaller orchestra like say an early Mozart symphony)) and likewise if I see a page with more systems than the rest, I'll add a page break again to even it out (more likely in a piano piece), like, after I've inputted all the notes of that particular section obviously, but it's generally mid-score that I do that kind of evening out unless the score is like only 1 section and that's it (which is most often not the case unless it's like a dance or a standalone fugue or something like that, most scores I work on are at least multi-section (like for example the aria A Berenice K 61c that I'll be starting on today which has both a Recitative section and an Aria section) if not multi-movement.

But yeah, I generally want the same number of systems per page throughout a given piece or movement, like all pages of a piano sonata at 6 systems a page or string quartet at 4 systems a page, that kind of thing, and usually I find the staff space adjustment to be enough for that, but sometimes there are those few pages where I need to adjust further.

**Comment by oktophonie** [comment]
Right now you can cram any number of bars on a system by applying the 'keep measures on same system to all bars on the system (except for the last, which should have a system or page break on it).

That's an absolutely horrible way to do it but it works for now. Improvements to managing systems are coming in 4.5, which should render this method obsolete.

**Comment by rpatters1** [comment]
Thank you for that!

Is there a GitHub issue or other publich posting that I can read that describes the coming changes in system management for 4.5?

**Comment by pacebes** [comment]
HI

I would like to understand the scope of this issue. Is it about adding a new line to this menu like "Add/remove page breaks"

![Image](https://github.com/user-attachments/assets/52a39f22-8e54-4e62-945d-196d9afd1f01)

with similar options to this page:

![Image](https://github.com/user-attachments/assets/86bf4e66-1f8f-4ba2-85ae-59b57c5d2432)

It could be:
- Break page every XXX systems
- Add page break at the end of each page
- Remove current page breaks

Or is it something similar to which is coming in 4.5 ? 

![Image](https://github.com/user-attachments/assets/3bc5f366-f089-4b35-93b6-8ba1780f84dc)

I don't know if 4.5 supports "page locks" or similar.

**Comment by wizofaus** [comment]
Yes, I had in mind the former. I don't know what System Locks are or whether there's a plan for "Page Locks".

**Comment by pacebes** [comment]
> Yes, I had in mind the former. I don't know what System Locks are or whether there's a plan for "Page Locks".

Thanks. I  will try to implement this way.

**Comment by pacebes** [comment]
I have created a PR to implement this issue

**Comment by pacebes** [comment]
Sorry to bother.

Could this issue be assigned to me and could anyone check if the PR created works as expected ?

Kind Regards.

**Comment by pacebes** [comment]
I have just created a [PR](https://github.com/musescore/MuseScore/pull/30597) from the latest version

**Comment by XiaoMigros** [comment]
Is this really a necessary feature to be added to the main application? The use cases seem extremely limited. My [evenly distribute systems across pages plugin](https://github.com/XiaoMigros/System-Spacing-Tools/) offers far more useful controls anyhow, without being overly complicated.

**Comment by XiaoMigros** [comment]
Especially since we can't yet lock the number of systems per page, like we can for measures per system...

**Comment by DaddyLudwig** [comment]
> Especially since we can't yet lock the number of systems per page, like we can for measures per system...

Yeah, that systems per page lock would be useful for me, because, as someone who inputs classical music into MuseScore, yes, I need the initial systems per page count to be flexible to have it settle somewhere. But once it does mostly settle on a consistent number of systems per page, there are undoubtedly going to be outliers, either pages with fewer systems than the average systems per page count or those with more. If it's mid-score, in the case of fewer, I nudge down the staff spacing until it evens out. In the case of more, I add page breaks until it's even. A systems per page lock would help with the latter case, once it's all x systems or more (where x is the average systems per page). Then I could just lock it at x systems per page, because I like having that systems per page consistency with the possible exception of the last page for obvious reasons and sometimes the first if evening out the first with the rest would make things too small (i.e. the same number of systems per page for a piano sonata or a string quartet or whatever)
